### PR TITLE
Add an `asBob` subtest to `virtualfund-single-hop_test.go`

### PR DESCRIPTION
**Pull request by geoknee**
Currently, we only test the virtual fund protocol over a single hop (3 participants in total), and never as "bob".  Thus there may be bugs in the code which are not surfaced yet. 

This approach adds such a test, and does so by:

- copy-pasting the "as Alice" test 
- making modifications so that it reflects Bob's situation
- ~fixing a bug in the `virtualfund` protocol where we used `n+1` instead of `n`~ this was cherry-picked off to #256 

This is a good approach because it solves the problem in a very verbose way that can be easily checked line-by-line. The downside is that the test is very repetitive and a little tiresome to read through. It is not easy to see the essential differences to the `asAlice` test.  I propose that we tackle refactoring the test (to DRY it out) in a future issue/PR. 

Additionally, I:
- Fixed a bug that confused `myRole` with `myIndex`. Easily done!
- tidied up some test types and functions that polluted the package

Towards #218.

As the change set is quite large, at the request of the reviewer I can split up the additions `TestNew`, `TestUpdate` and `TestCrank`. 

---

#### Code quality

- [x] I have written clear commit messages
- [ ] I have performed a self-review of my own code
- [ ] This change does not have an unduly wide scope
- [x] I have separated logic changes from refactor changes (formatting, renames, etc.)
- [x] I have commented my code wherever necessary (can be 0)
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary

#### Project management

- [x] I have assigned myself to this PR
- [ ] I have linked the appropriate github issue

### CODE REVIEW

**Comment by geoknee** [comment]
I think [105567f](https://github.com/statechannels/go-nitro/pull/255/commits/105567f15f04740c645e6c5750a2990f446e6190) is misguided / wrong. I think the confusion is slightly subtler, and have marked this as a draft until I fix it.

**Comment by geoknee** [comment]
I have cherry-picked the fix to `n` onto #256 and will get this PR ready for review after that has been merged.

File: protocols/virtualfund/virtualfund-single-hop_test.go [hunk 1]
```diff
@@ -14,26 +14,28 @@ import (
...
+		address     types.Address
+		destination types.Destination
+		privateKey  []byte
+		role        uint
 	}
-	err = ss.AddSignature(sig)
-	if err != nil {
-		panic(err)
+
+	signState := func(s state.State, a actor) state.SignedState {
```
Comment (lalexgap) [inline protocols/virtualfund/virtualfund-single-hop_test.go:26]: Not related to this PR but I think `signState` can be removed and we can use `ss.Sign` instead.

**Review by lalexgap — APPROVED** [review]
This looks good to me!
 
> I propose that we tackle refactoring the test (to DRY it out) in a future issue/PR.

I'd be in favour of doing this soon! It's already getting quite large and difficult to follow.

**Pull request merged**